src/erlcloud_s3.erl

```diff
@@ -315,16 +315,20 @@ delete_objects_batch_timeout(#aws_config{timeout = Timeout}) ->
 %    ok
 %
 -spec explore_dirstructure(string(), list(), list()) -> list().
+explore_dirstructure(Bucketname, Branches, Accum) ->
+    explore_dirstructure(Bucketname, Branches, Accum, default_config()).
 
-explore_dirstructure(_, [], Result) ->
+-spec explore_dirstructure(string(), list(), list(), aws_config()) -> list().
+explore_dirstructure(_, [], Result, _Config) ->
                                     lists:append(Result);
-explore_dirstructure(Bucketname, [Branch|Tail], Accum) ->
+explore_dirstructure(Bucketname, [Branch|Tail], Accum, Config)
+    when is_record(Config, aws_config) ->
     ProcessContent = fun(Data)->
             Content = proplists:get_value(contents, Data),
             lists:foldl(fun(I,Acc)-> R = proplists:get_value(key, I), [R|Acc] end, [], Content)
             end,
 
-    Data = erlcloud_s3:list_objects(Bucketname,[{prefix, Branch}, {delimiter, "/"}]),
+    Data = erlcloud_s3:list_objects(Bucketname, [{prefix, Branch}, {delimiter, "/"}], Config),
```

Comment:
Since you're here, it seems inappropriate to call `erlcloud_s3` as an external module from within itself. Can you remove that?